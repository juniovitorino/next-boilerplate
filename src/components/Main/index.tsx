import * as st from './styles'

const Main = ({ title = 'React AvanÃ§ado' }) => (
  <main>
    <st.Title>{title}</st.Title>
  </main>
)

export default Main
