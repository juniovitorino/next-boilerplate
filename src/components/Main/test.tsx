import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)
    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()
  })

  it('should render h1 heading', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
