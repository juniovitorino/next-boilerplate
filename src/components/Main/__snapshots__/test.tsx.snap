// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render h1 heading 1`] = `
<main>
  <h2>
    React Avan√ßado
  </h2>
</main>
`
